src/bin/demo.rs

```diff
@@ -374,6 +463,62 @@ fn main() -> anyhow::Result<()> {
         );
     }
 
+    // --------------------------------------------------------------------
```

Comment:
The `f2_tx` should use `build_script_f2_blake3_locked` to enforce `x<F2_THRESHOLD`. Then spending transaction should present the witness for `f2_tx` locking script.